### PR TITLE
Run dotnet format on the entire codebase (and enforce, going forward)

### PR DESCRIPTION
**Pull request by mikeage**
Add `dotnet-format` to the checkers run by pre-commit.

By running `pip install pre-commit && pre-commit install` in your clone (python3 required), it will automatically run any newly modified files through `dotnet-format`. Additionally, as the style rules are in `.editorconfig`, they should be respected by most editors.

Once this is merged, the commit can be added to a .git-blame-ignore-revs file so that `git blame` from the command line will skip any changes made as part of this PR. See, for example, https://www.moxio.com/blog/43/ignoring-bulk-change-commits-with-git-blame for more information

### CODE REVIEW

**Comment by mikeage** [comment]
Marked as draft since we don't want to merge this while we have open PRs

**Comment by mikeage** [comment]
Note that the diff is *huge* because there's quite a lot of whitespace changes, due to some seemingly missed indentation (or maybe I didn't understand the style guide used, that's also quite possible!). 

To see the changes without the whitespace, please look at https://github.com/icosa-gallery/open-brush/pull/81/files?w=1 . There might be additional options we can use to keep the current style; I'm not an expert in C#.

File: .editorconfig [hunk 1]
```diff
@@ -7,3 +7,6 @@ root = true
 indent_style = space
 indent_size = 2
 
+# Based on Google's C# style guide
```
Comment (billyquith) [inline .editorconfig:10]: 4 space indents will make this comment no longer true!

Comment (mikeage) [reply .editorconfig:10]: Good point, this is from an earlier version where I tried to mimic the original style

**Pull request merged**